src/builder.rs

```diff
@@ -0,0 +1,82 @@
+use std::sync::Arc;
+
+use blockstore::Blockstore;
+
+use crate::client::{ClientBehaviour, ClientConfig};
+use crate::multihasher::{Multihasher, MultihasherTable};
+use crate::utils::stream_protocol;
+use crate::{BitswapBehaviour, Result};
+
+/// Builder for [`BitswapBehaviour`].
+pub struct BitswapBehaviourBuilder<const S: usize, B>
+where
+    B: Blockstore + Send + Sync + 'static,
+{
+    protocol_prefix: Option<String>,
+    blockstore: B,
+    client: ClientConfig,
+    multihasher: MultihasherTable<S>,
+}
+
+impl<const S: usize, B> BitswapBehaviourBuilder<S, B>
+where
+    B: Blockstore + Send + Sync + 'static,
+{
+    /// Creates a new builder for [`BitswapBehaviour`].
+    pub fn new(blockstore: B) -> Self {
+        BitswapBehaviourBuilder {
+            protocol_prefix: None,
+            blockstore,
+            client: ClientConfig {
+                set_send_dont_have: true,
+            },
+            multihasher: MultihasherTable::<S>::new(),
+        }
+    }
+
+    /// Set a prefix on the protocol name.
+    ///
+    /// The prefix will be added on `/ipfs/bitswap/1.2.0` and it must start with `/`.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// # use blockstore::InMemoryBlockstore;
+    /// BitswapBehaviour::builder(InMemoryBlockstore::new())
+    ///     .protocol_prefix("/celestia/celestia")
+    ///     .build()?;
+    /// ```
+    pub fn protocol_prefix(&mut self, prefix: &str) {
```

Comment:
shouldn't those take and return the builder to use it like in doc?

Reply:
yeah i forgot to do it